homeassistant/components/group/sensor.py

```diff
@@ -377,59 +406,65 @@ def icon(self) -> str | None:
             return "mdi:calculator"
         return None
 
-    @property
-    def state_class(self) -> SensorStateClass | str | None:
-        """Return state class."""
-        if self._attr_state_class is not None:
-            return self._attr_state_class
-        return self.calc_state_class
+    def _calculate_state_class(
+        self, state_class: SensorStateClass | None
+    ) -> SensorStateClass | None:
+        """Calculate state class."""
+        if state_class:
+            return state_class
+        state_classes: list[SensorStateClass | None] = []
+        for entity_id in self._entity_ids:
+            try:
+                _state_class = get_capability(self.hass, entity_id, "state_class")
+            except HomeAssistantError:
+                return None
+            if not _state_class:
+                return None
+            state_classes.append(_state_class)
+
+        if all(x == state_classes[0] for x in state_classes):
+            return state_classes[0]
+        return None
```

Comment:
Should we maybe log a warning or create a repair issue if state classes are conflicting? Or it's OK? Same question for device class.

Reply:
I was thinking what if the user wants to combine for sensors with conflicting. On the other hand that user could simply ignore the repair issue so I guess it makes sense for the most of the use-cases to implement it 👍